R/plotLoadings.R

```diff
@@ -0,0 +1,362 @@
+                        axis.text.x = element_blank()) +
+                    scale_fill_gradient2(limits = c(-1,1), low = "darkslateblue",
+                        mid = "white", high = "darkred", guide = NULL) +
+                    geom_text(color="black", size=4 - 0.075 * n) +
+                    labs(title=names[i])
+            # Only show legend for last one
+            } else {
+                p <- p +
+                    ggplot(df, aes(x = "PC", y = Feature, label = PC))  +
+                    geom_point(aes(fill = PC), size=15 - 0.4*n, shape = 22) +
+                    theme(axis.title.x = element_blank(), axis.title.y = element_blank(),
+                        axis.text.x = element_blank()) +
+                    scale_fill_gradient2(limits = c(-1,1), low = "darkslateblue",
+                        mid = "white", high = "darkred") +
+                    geom_text(color="black", size=4 - 0.075*n) +
+                    labs(title=names[i])
+            }
+        }
+    }
+    
+    else if (layout == "barplot") {
+        # Transform into a dataframe
+        df <- data.frame(PC = L[[1]][[1]], Feature = L[[1]][["Feature"]]) 
+        # Plot first component
+        p <- ggplot(df, aes(x = .data[["PC"]], .data[["Feature"]])) +
+            geom_bar(stat="identity") +
+            theme(axis.title.x = element_blank()) +
+            xlim(-1,1) +
+            labs(title=names[1])
+        # Loop to plot others components
+        for (i in 2:length(L)) {
+            df <- data.frame(PC =L[[i]][[i]],
+                Feature = L[[i]][["Feature"]])
+            p <- p + ggplot(df, aes(x = .data[["PC"]], .data[["Feature"]])) +
+                geom_bar(stat="identity") +
+                theme(axis.title.x = element_blank()) +
+                xlim(-1,1) +
+                labs(title=names[i])
+        }
+    }
+    
+    else if (layout == "screeplot") {
+        # Transform into a dataframe
+        df <- data.frame(PC = L[[1]][[1]], Feature = L[[1]][["Feature"]]) 
+        # Plot first component
+        p <- ggplot(df, aes(x = .data[["PC"]], .data[["Feature"]])) +
+            geom_bar(stat="identity") +
+            theme(axis.title.x = element_blank(), 
+                axis.text.x = element_text(angle = 90, 
+                    , size = 6, vjust = 0.5, hjust=1)) +
+            xlim(-1,1) +
+            labs(title=names[1]) +
+            coord_flip()
+        # Plot others components
+        for (i in 2:length(L)) {
+            df <- data.frame(PC =L[[i]][[i]],
+                Feature = L[[i]][["Feature"]])
+            p <- p + ggplot(df, aes(x = .data[["PC"]], .data[["Feature"]])) +
+                geom_bar(stat="identity") +
+                theme(axis.title.x = element_blank(),
+                    axis.text.x = element_text(angle = 90,
+                        vjust = 0.5, hjust=1, size=6)) +
+                xlim(-1,1) +
+                labs(title=names[i]) +
+                coord_flip()
+        }
```

Comment:
Instead of looping like this, you could use facet_by which is more neat solution (then all the data must be in same data.frame)

Reply:
```
# Prepare the data in correct format, this might not take into account
# the correct order and features, but it gives the idea
df <- lapply(seq_len(length(L)), function(i){
    L[[i]][, i, drop = FALSE]
})
df <- do.call(cbind, df) 
df[["feat"]] <- rownames(df)

# To long format
df <- reshape(
    df, varying =colnames(df)[ !colnames(df) %in% c("feat") ],
    v.names = "value", 
    timevar = "PC",
    times = colnames(df)[ !colnames(df) %in% c("feat") ],
    direction = "long")

ggplot(df, aes(x = .data[["value"]], y = .data[["feat"]])) +
    geom_bar(stat="identity") +
    facet_wrap(~ PC, scales = "free")

```

Reply:
Check also other plot types. I think the data.frame list might not be needed. Just use facets instead of looping over list of data.frames

Reply:
This is indeed a nice solution but after trying for hours, haven't been able to make axes label change for every component like it used to be. Also ordering seems to be working only on first component for some reason. 
![image](https://github.com/user-attachments/assets/a21f8658-2f1d-40d7-b436-97e51088e1ba)
Here is the best I have been able to do but I guess it looks worse

Reply:
I think you would need to define every (sub)figure separately in order to have a full control.

Reply:
This works for heatmap
```

library(tidytext)
library(shadowtext)
ggplot(df, aes(x = .data[["PC"]], y = reorder_within(Feature, Value, PC), label = round(.data[["Value"]], 2)))  +
    geom_point(aes(fill = .data[["Value"]]), size = 15-0.4*n, shape = 22) +
    scale_fill_gradient2(
        limits = c(-1,1), low = "darkslateblue", mid = "white", high = "darkred") +
    scale_y_reordered() +
    geom_shadowtext(size = 4-0.075*n) +
    facet_wrap(~ PC, scales = "free")
```